templates/k8s/job-build.jinja2

```diff
@@ -0,0 +1,71 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ job_name }}
+spec:
+  backoffLimit: 3
+  template:
+    spec:
+      restartPolicy: OnFailure
+
+      # in-memory tmpfs
+      volumes:
+      - name: scratch-volume
+        emptyDir: { medium: "Memory" }
+
+      # only run on preemptible nods
+      #nodeSelector:
+      #  cloud.google.com/gke-preemptible: "true"
+
+      containers:
+      - name: kci-build
+        image: {{ "FIXME" | env_override('DOCKER_BASE') }}build-{{ "FIXME" | env_override('DOCKER_IMAGE') }}
+
+        volumeMounts:
+        - mountPath: "/scratch"
+          name: scratch-volume
+
+        command: ["/bin/bash", "-x", "-c"]
+        args: ["echo nproc=$(nproc) && df && free && \
+export KDIR=/scratch/linux && export OUTPUT=$HOME/build && export CCACHE_DISABLE=true && env &&\
+git clone --depth=1 --branch ${KCI_CORE_BRANCH} ${KCI_CORE_URL} && cd kernelci-core && \
+./kci_build pull_tarball --kdir ${KDIR} --url ${SRC_TARBALL} --retries 3 --delete && \
```

Comment:
> Does this mean there would have to be a builder with "k8s" label, which in turn launches nodes in kubernetes?
> Can 1 builder launch many kubernetes builds or is it a 1:1 ratio?

If you use the official k8s jenkins plugin, it's a 1:1 ration, but I'm not using that (for many lots of reasons.)

In this approach, the builder is just to drive k8s via the cmdline (kubectl), so the builder node can launch as many k8s builds as we want.

My idea was that we create a single `k8s` builder with the #executors set to the maximim number of builds we can support in the k8s cluster.

Reply:
> It might be good to split the downloads into a Kubernetes init step. It looks like this lets us take advantage of Kubernetes' retry mechanisms - I know there's some in kci_build itself but it's a bit more idiomatic.

I'm new enough to k8s to not be sure how to do that.  AFAICT, each step in k8s is a separate container, and while you can then chain them with dependencies, I haven't looked closely at how to do that yet.  Hopefully someone who knows more about k8s can help us evolve this in an intelligent way.

Reply:
It's pretty noddy to use - the examples in the page linked below are good, it's basically just a list of containers/commands that it steps through in series in the order they're listed in the spec. I'm using it by putting the /scratch volume in the initContainers and doing the source fetches there.

https://kubernetes.io/docs/concepts/workloads/pods/init-containers/

Reply:
> > Does this mean there would have to be a builder with "k8s" label, which in turn launches nodes in kubernetes?
> > Can 1 builder launch many kubernetes builds or is it a 1:1 ratio?
> 
> If you use the official k8s jenkins plugin, it's a 1:1 ration, but I'm not using that (for many lots of reasons.)
> 
> In this approach, the builder is just to drive k8s via the cmdline (kubectl), so the builder node can launch as many k8s builds as we want.
> 
> My idea was that we create a single `k8s` builder with the #executors set to the maximim number of builds we can support in the k8s cluster.

Ah cool, so really it's just a very low cpu k8s "monitor" job, gotcha